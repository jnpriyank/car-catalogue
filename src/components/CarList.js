import React, { useState, useEffect } from "react";
import Car from "./car";
import CategoryList from "./categoryList";

const CarList = (props) => {
  const [selectedCategory, setSelectedCategory] = useState(
    props.categories ? props.categories[0].name : ""
  );

  const onCategoryChangeHandler = (category) => {
    setSelectedCategory(category);
  };

  if (!props.categories || props.categories.length === 0) {
    return null;
  }

  return (
    <div className="car-list-container">
      <header className="header">
        <h2 className="item">Car</h2>
        <h1 className="item">Categories</h1>
        <div className="item nav-categories">
          <CategoryList
            categories={props.categories}
            selectedCategory={selectedCategory}
            onCategoryChange={onCategoryChangeHandler}
          ></CategoryList>
        </div>
      </header>
      <section className="body">
        {props.cars
          .filter((car) => car.category === selectedCategory)
          .map((value) => {
            return (
              <Car
                key={value.id}
                name={value.name}
                imageName={value.imageName}
              ></Car>
            );
          })}
      </section>
    </div>
  );
};

export default CarList;
