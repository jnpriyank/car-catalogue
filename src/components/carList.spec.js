import React from "react";
import { shallow } from "enzyme";
import CarList from "./carList";
import categoriesData from "../data/categories";
import carsData from "../data/cars";

describe("car list component", () => {
    it("should be rendered successfully, when given proper data", () => {
        // shallow render the component with the data
        const wrapper = shallow(<CarList categories={categoriesData} cars={carsData}></CarList>);
        expect(wrapper.find(".car-list-container")).toHaveLength(1);
        
        // perform category related assertions
        const categoryListComponent = wrapper.find(".nav-categories").find("CategoryList");
        expect(categoryListComponent).toHaveLength(1);
        expect(categoryListComponent.prop("categories")).toHaveLength(3);
        expect(categoryListComponent.prop("selectedCategory")).toEqual("Hatchbacks");
        expect(categoryListComponent).toMatchInlineSnapshot(`
<CategoryList
  categories={
    Array [
      Object {
        "description": "Hatchbacks description",
        "id": 1,
        "name": "Hatchbacks",
      },
      Object {
        "description": "Vans description",
        "id": 2,
        "name": "Vans",
      },
      Object {
        "description": "Sedans description",
        "id": 3,
        "name": "Sedans",
      },
    ]
  }
  onCategoryChange={[Function]}
  selectedCategory="Hatchbacks"
/>
`);

// perform car related assertions
expect(wrapper.find(".body Car")).toHaveLength(4);
expect(wrapper.find(".body Car")).toMatchInlineSnapshot(`
Array [
  <Car
    imageName="wagonr.png"
    key="1"
    name="Wagon R"
  />,
  <Car
    imageName="800.png"
    key="2"
    name="800"
  />,
  <Car
    imageName="celerio.png"
    key="3"
    name="celerio"
  />,
  <Car
    imageName="swift.png"
    key="4"
    name="swift"
  />,
]
`);
    });

    it("should not be rendered if there are no categories", () => {
        const wrapper = shallow(<CarList cars={carsData}></CarList>);
        expect(wrapper.find(".car-list-container")).toHaveLength(0);
        expect(wrapper).toMatchSnapshot();
    });
});