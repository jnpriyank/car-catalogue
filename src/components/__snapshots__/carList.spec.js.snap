// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`car list component should not be rendered if there are no categories 1`] = `""`;
